crates/evm/core/src/utils.rs

```diff
@@ -109,43 +109,89 @@ pub fn configure_tx_env(env: &mut revm::primitives::Env, tx: &Transaction<AnyTxE
     }
 }
 
+fn field_to_val<T: FromStr + Default>(
+    fields_map: &BTreeMap<String, serde_json::Value>,
+    field_name: &str,
+) -> T {
+    fields_map
+        .get(field_name)
+        .and_then(|v| v.as_str())
+        .and_then(|s| T::from_str(s).ok())
+        .unwrap_or_default()
+}
+
+fn field_to_bytes(
+    fields_map: &BTreeMap<String, serde_json::Value>,
+    field_name: &str,
+) -> alloy_primitives::Bytes {
+    fields_map
+        .get(field_name)
+        .and_then(|v| v.as_str())
+        .and_then(|s| alloy_primitives::hex::decode(s).ok())
+        .map(Into::into) // Convert Vec<u8> to Bytes
+        .unwrap_or_default()
+}
+
 /// Configures the env for the given RPC transaction.
 /// Accounts for an impersonated transaction by resetting the `env.tx.caller` field to `tx.from`.
 pub fn configure_zksync_tx_env(
     env: &mut revm::primitives::Env,
-    tx: &Transaction<AnyTxEnvelope>,
+    outer_tx: &Transaction<AnyTxEnvelope>,
 ) -> foundry_zksync_core::ZkTransactionMetadata {
     let mut metadata = foundry_zksync_core::ZkTransactionMetadata::default();
-    if let AnyTxEnvelope::Unknown(tx) = &tx.inner.inner() {
-        let input = tx
+
+    if let AnyTxEnvelope::Unknown(unknown_tx_data) = &outer_tx.inner.inner() {
+        let input_bytes_from_fields = unknown_tx_data
             .inner
             .fields
             .get("input")
             .and_then(|value| value.as_str())
             .and_then(|value| alloy_primitives::hex::decode(value).ok())
             .unwrap_or_default();
-        let (eip712_tx, _) =
-            zksync_types::transaction_request::TransactionRequest::from_bytes_unverified(&input)
-                .expect("invalid zksync transaction");
-
-        env.tx.transact_to = TxKind::Call(eip712_tx.to.expect("to must exist").to_address());
-        env.tx.caller = eip712_tx.from.expect("from must exist").to_address();
-        env.tx.gas_limit = eip712_tx.gas.as_u64();
-        env.tx.nonce = Some(eip712_tx.nonce.as_u64());
-        env.tx.value = eip712_tx.value.to_ru256();
-        env.tx.data = eip712_tx.input.0.into();
-        env.tx.chain_id = eip712_tx.chain_id;
-        env.tx.gas_price = eip712_tx.gas_price.to_ru256();
-        env.tx.gas_priority_fee = eip712_tx.max_priority_fee_per_gas.map(|value| value.to_ru256());
-
-        if let Some(eip712_meta) = eip712_tx.eip712_meta {
-            metadata = foundry_zksync_core::ZkTransactionMetadata {
-                factory_deps: eip712_meta.factory_deps,
-                paymaster_data: eip712_meta.paymaster_params,
-            };
+
+        // Note(ZK): Sometimes the input retrieved does not contain the encoded transaction. This
+        // happens mainly in sepolia and mainnet but not in anvil-zksync. That is why we
+        // need to check if the input is a valid transaction request and if not, we use the fields
+        // from the tx.
```

Comment:
This is a bit misleading, the actual reason is that `anvil-zksync` is potentially indexing transactions via `tx_hash` as raw transactions, where it should be mimicking what happens on-chain. 

As the initial feature was develope with `anvil-zksync` - this was assumed to be the default behavior. We might wanna look into anvil-zksync if this behavior needs to be fixed \cc @dutterbutter